src/dlio/odom.cc

```diff
@@ -385,14 +392,16 @@ void dlio::OdomNode::publishToROS(pcl::PointCloud<PointType>::ConstPtr published
   transformStamped.header.frame_id = this->odom_frame;
   transformStamped.child_frame_id = this->baselink_frame;
 
-  transformStamped.transform.translation.x = this->state.p[0];
-  transformStamped.transform.translation.y = this->state.p[1];
-  transformStamped.transform.translation.z = this->state.p[2];
+  publishedPoseMtx.lock();
+  transformStamped.transform.translation.x = this->publishedPose.p[0];
+  transformStamped.transform.translation.y = this->publishedPose.p[1];
+  transformStamped.transform.translation.z = this->publishedPose.p[2];
```

Comment:
The data used here could still be different from the poseCopy variable, resulting in a mismatch